scripts/snippets/agribalyse_categories_csv.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+"""A small script to export agribalyse categories in a CSV using categories.full.json
+"""
+import json
+import csv
+
+datas = json.load(open("categories.full.json", "r"))
+
+
+# get agribalyse
+all_agri_cats = set(tagid for tagid, d in datas.items() for k in d.keys() if "agribalyse" in k)
+agri_cats = all_agri_cats
+
+# and children (until we have no more children)
+while agri_cats:
+    children = set(
+        tagid
+        for tagid, d in datas.items()
+        if set(d.get("parents",[])) & agri_cats
+    )
+    all_agri_cats |= children
+    agri_cats = children
+
+# get agribalyse exploring parents
+def agribalyse(tagid, depth=0):
```

Comment:
I would use function annotation there: https://peps.python.org/pep-0008/#function-annotations
`def agribalyse(tafid: srtr, depth: int = 0)  -> (str, int): `
and then, return a str and a int all the time